Panic when `cpus` omitted from multipass_instance

I am new to Terraform, and I am running on an M1 Mac, so I wanted to try out multipass to easily provision a local VM.

I created a very simple terraform file:

```console
$ cat main.tf
terraform {
  required_providers {
    multipass = {
      source  = "larstobi/multipass"
      version = "~> 1.4.0"
    }
  }
}

provider "multipass" {}

resource "multipass_instance" "test1" {
  name = "test1"
  disk = "512MiB"
  memory = "512MiB"
}
```

Then I ran `terraform init` and `terraform plan` successfully. But `terraform apply` panicked:

```console
$ terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # multipass_instance.test1 will be created
  + resource "multipass_instance" "test1" {
      + disk   = "512MiB"
      + memory = "512MiB"
      + name   = "test1"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

multipass_instance.test1: Creating...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-multipass_v1.4.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x7 pc=0x1044210ac]

goroutine 26 [running]:
math/big.(*Float).Append(0x0, {0x14000130210?, 0x0?, 0x0?}, 0x67, 0xa)
	math/big/ftoa.go:65 +0x2c
math/big.(*Float).Text(...)
	math/big/ftoa.go:52
math/big.(*Float).String(0x104a52398?)
	math/big/ftoa.go:58 +0x50
terraform-provider-multipass/internal/provider.instanceResource.Create({{_, {_, _}}}, {_, _}, {{{{0x104a550a0, 0x140003cf440}, {0x10498ef20, 0x140003cef90}}, {0x14000189b90, ...}}, ...}, ...)
	terraform-provider-multipass/internal/provider/resource_instance.go:125 +0x254
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x14000334d80, {0x104a52398, 0x140003ceb40}, 0x14000793360, 0x140007932a8)
	github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/fwserver/server_createresource.go:81 +0x3a0
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x104a52398?, {0x104a52398, 0x140003ceb40}, 0x14000284ab0, 0x14000793540)
	github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/fwserver/server_applyresourcechange.go:51 +0x408
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x14000334d80, {0x104a52398?, 0x140003ce9f0?}, 0x14000218550)
	github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/proto6server/server_applyresourcechange.go:52 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000310500, {0x104a52398?, 0x140003ce210?}, 0x14000124150)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov6/tf6server/server.go:811 +0x370
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x104a29e60?, 0x14000310500}, {0x104a52398, 0x140003ce210}, 0x140001240e0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:385 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002ee1e0, {0x104a55200, 0x140000036c0}, 0x14000334000, 0x140003cfb90, 0x104e25fe0, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0x9d8
google.golang.org/grpc.(*Server).handleStream(0x140002ee1e0, {0x104a55200, 0x140000036c0}, 0x14000334000, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x298

Error: The terraform-provider-multipass_v1.4.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I looked at the line that panicked and saw it was the `cpus` value. According to the [multipass docs](https://registry.terraform.io/providers/larstobi/multipass/latest/docs/resources/instance), `cpus` defaults to 1 so I omitted it. Adding the line `cpus = 1` fixed that issue (but it failed again due to my choice of value for `disk`, which I easily addressed).

--- a/internal/provider/resource_instance.go
+++ b/internal/provider/resource_instance.go
@@ -119,10 +119,17 @@ func (r instanceResource) Create(ctx context.Context, req tfsdk.CreateResourceRe
         "name": plan.Name.String(),
     })
 
+    var cpus string
+    if plan.CPUS.Null {
+        cpus = ""
+    } else {
+        cpus = plan.CPUS.Value.String()
+    }
+
     _, err := multipass.LaunchV2(&multipass.LaunchReqV2{
         Name:          plan.Name.Value,
         Image:         plan.Image.Value,
-        CPUS:          plan.CPUS.Value.String(),
+        CPUS:          cpus,
         Memory:        plan.Memory.Value,
         Disk:          plan.Disk.Value,
         CloudInitFile: plan.CloudInitFile.Value,

--- a/internal/provider/utils.go
+++ b/internal/provider/utils.go
@@ -18,32 +18,39 @@ func QueryInstance(state Instance) (*Instance, error) {
 	}
 
 	// Check CPUS
-	cpus := new(big.Float)
-	cpus.SetString(out.CPUS)
-
-	// Check Memory
-	if state.Memory.Null {
-		state.Memory = types.String{Value: "0MiB"}
+	current_cpus := state.CPUS
+	// If CPUS is not specified, then ignore changes
+	if !state.CPUS.Null {
+		cpus := new(big.Float)
+		cpus.SetString(out.CPUS)
+		if cpus != state.CPUS.Value {
+			current_cpus = types.Number{Value: cpus}
+		}
 	}
 
+	// Check Memory
 	current_memory := state.Memory
-	if equal, err := CompareDataSizes(out.Memory, state.Memory.Value); err != nil {
-		return nil, errors.New("Error comparing memory size: " + err.Error())
-	} else {
-		if !*equal {
-			current_memory = types.String{Value: RemoveZeroDecimalAndSpaces(out.Memory)}
+	// If Memory is not specified, then ignore changes
+	if !state.Memory.Null {
+		if equal, err := CompareDataSizes(out.Memory, state.Memory.Value); err != nil {
+			return nil, errors.New("Error comparing memory size: " + err.Error())
+		} else {
+			if !*equal {
+				current_memory = types.String{Value: RemoveZeroDecimalAndSpaces(out.Memory)}
+			}
 		}
 	}
 
 	// Check Disk
-	var current_disk types.String
-	if equal, err := CompareDataSizes(out.Disk, state.Disk.Value); err != nil {
-		return nil, errors.New("Error comparing disk size: " + err.Error())
-	} else {
-		if *equal {
-			current_disk = state.Disk
+	current_disk := state.Disk
+	// If Disk is not specified, then ignore changes
+	if !state.Disk.Null {
+		if equal, err := CompareDataSizes(out.Disk, state.Disk.Value); err != nil {
+			return nil, errors.New("Error comparing disk size: " + err.Error())
 		} else {
-			current_disk = types.String{Value: RemoveZeroDecimalAndSpaces(out.Disk)}
+			if !*equal {
+				current_disk = types.String{Value: RemoveZeroDecimalAndSpaces(out.Disk)}
+			}
 		}
 	}
 
@@ -51,7 +58,7 @@ func QueryInstance(state Instance) (*Instance, error) {
 	var result = Instance{
 		Name:          state.Name,
 		Image:         state.Image,
-		CPUS:          types.Number{Value: cpus},
+		CPUS:          current_cpus,
 		Memory:        current_memory,
 		Disk:          current_disk,
 		CloudInitFile: state.CloudInitFile,

